src/main/java/com/google/research/bleth/simulator/RandomAwakenessStrategy.java

```diff
@@ -0,0 +1,24 @@
+package com.google.research.bleth.simulator;
+
+import java.util.Random;
+
+/** A random awakeness strategy for an observer - the observer wakes up at random time each awakeness cycle. */
+public class RandomAwakenessStrategy implements AwakenessStrategy {
+    private Random random = new Random();
+
+    /**
+     * Determine when the observer wakes up next randomly within the next awakeness cycle.
+     * @param nextIntervalStart is the first round in which the observer can wake up again.
+     * @param awakenessCycle is the duration of each awakeness cycle - an observer can wake up once such cycle.
+     * @param awakenessDuration is the number of rounds that the observer is awake each time it wakes up.
+     * @param lastAwakeningTime is the last round the observer woke up.
+     * @return the next round in which the observer wakes up.
+     */
+    @Override
+    public int nextTime(int nextIntervalStart, int awakenessCycle, int awakenessDuration, int lastAwakeningTime) {
+        if (awakenessCycle == awakenessDuration) {
+            return 0;
```

Comment:
Is this correct? You can return a value below nextIntervalStart - seems to me like this will break the logic in updateAwakenessState?
(Consider if this may mean something for the API: e.g., maybe the awakeness strategy should return a positive sleep duration, rather than an absolute wakeup cycle?)